corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -224,22 +225,44 @@ def get_docs(self, from_db, startkey, last_doc_pk=None, limit=500):
         """
         raise NotImplementedError
 
+    @property
+    def count_query(self):
+        """Define a custom query to use to get the doc count (only used for filtered reindex.
+        :returns: tuple('query_string', [query values])
+        """
+        return
+
     def get_doc_count(self, from_db):
         """Get the doc count from the given DB
         :param from_db: The DB alias to query
         """
-        from_db = 'default' if from_db is None else from_db
-        sql_query = "SELECT reltuples FROM pg_class WHERE oid = '{}'::regclass"
-        db_cursor = connections[from_db].cursor()
-        with db_cursor as cursor:
-            cursor.execute(sql_query.format(self.model_class._meta.db_table))
-            return int(fetchone_as_namedtuple(cursor).reltuples)
+        if not self.domain:
```

Comment:
they could both be done that way but the non-domain way is likely to be a bit more accurate. Not super important in this case so we could just do it the one way.